### PR TITLE
Name res cleanup, docs and inline rules

### PR DESCRIPTION
**Pull request by max-hoffman**
Move bespoke rules into planbuilder. Functions are only ran for their necessary nodes. Also adds copyright headers.

Requires some Dolt changes: https://github.com/dolthub/dolt/pull/6567

### CODE REVIEW

**Review by fulghum — APPROVED** [review]
LGTM! 🚢

File: sql/planbuilder/builder.go [hunk 1]
```diff
@@ -44,11 +59,16 @@ type ProcContext struct {
 }
 
 func New(ctx *sql.Context, cat sql.Catalog) (*Builder, error) {
+	opts := ast.ParserOptions{AnsiQuotes: false}
 	sqlMode, err := sql.LoadSqlMode(ctx)
-	if err != nil {
-		return nil, err
+	if err == nil {
```
[the remark this answers is not in the document]
Comment (max-hoffman) [reply sql/planbuilder/builder.go:64]: yep good call. in this case I assume if system variables have not been initialized, we should use the default value

[the remark this answers is not in the document]
Comment (max-hoffman) [reply sql/planbuilder/builder.go:64]: i guess i could move this into sql.LoadSqlMode, alternatively

File: sql/planbuilder/show.go [hunk 2]
```diff
@@ -618,9 +632,20 @@ func (b *Builder) buildShowAllColumns(inScope *scope, s *ast.Show) (outScope *sc
...
+		if err != nil {
+			b.handleErr(err)
+		}
+	default:
+	}
 	if rt, _ := table.(*plan.ResolvedTable); rt != nil {
-		show.Indexes = b.getInfoSchemaIndexes(rt)
-		node = b.modifySchemaTarget(tableScope, show, rt)
+
 	}
```
[the remark this answers is not in the document]
Comment (max-hoffman) [reply sql/planbuilder/show.go]: good call, deleted

**Pull request merged**